src/sentry_os.c

```diff
@@ -0,0 +1,175 @@
+#include "sentry_os.h"
+#include "sentry_string.h"
+
+#ifdef SENTRY_PLATFORM_WINDOWS
+
+#    include <winver.h>
+
+sentry_value_t
+sentry__get_os_context(void)
+{
+    sentry_value_t os = sentry_value_new_object();
+    if (sentry_value_is_null(os)) {
+        return os;
+    }
+
+    sentry_value_set_by_key(os, "name", sentry_value_new_string("Windows"));
+
+    void *ffibuf = NULL;
+
+    DWORD size = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
```

Comment:
Why `kernel32.dll` version is used instead of `GetVersionExW()`?

Libraries versions does not match the Windows build version.

![image](https://user-images.githubusercontent.com/704382/106137071-41ef3700-617b-11eb-80fc-209d401b1fd6.png)

Reply:
Again, I copied whatever crashpad is doing, to be consistent with those tools. I was a bit surprised myself that the version does not reflect the 20H2 release number.

I will discuss this internally.

Reply:
Also, you can use `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion` (that's what winver actually use)

* Version: `<CurrentMajorVersionNumber>.<CurrentMinorVersionNumber>`
* Full Build `<CurrentMajorVersionNumber>.<CurrentMinorVersionNumber>.<CurrentBuildNumber>.<UBR>`
* Edition: `<EditionID>`
* Update name: `<DisplayVersion>`

Reply:
Also, you can combine `GetVersionExW` and registry to detect if application is running under compatibility mode.